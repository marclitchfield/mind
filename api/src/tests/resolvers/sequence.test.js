import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Sequence tests', () => {
  applySetup();

  test('create sequence', () => {
    return verifyMutations('Sequence tests', 'create sequence', {
      mutations: [
        ({space}) => gql`mutation { Person { person1:post_in_space(input:{sourceId:"${space.id}", title: "element 1"}) { id } } }`,
        ({space}) => gql`mutation { Person { person2:post_in_space(input:{sourceId:"${space.id}", title: "element 2"}) { id } } }`,
        ({space}) => gql`mutation { Concept { concept1:post_in_space(input:{sourceId:"${space.id}", title: "element 3"}) { id } } }`,        
        ({space}) => gql`mutation { Concept { concept2:post_in_space(input:{sourceId:"${space.id}", title: "element 4"}) { id } } }`,
        ({space}) => gql`mutation { Concept { type:post_in_space(input:{sourceId:"${space.id}", title: "sequence type"}) { id } } }`,
        ({type, person1}) => gql`mutation { Sequence { sequence:post_next_person(input:{sourceId:"${person1.id}", classId:"${type.id}", title:"test sequence"}) { id } } }`,
        ({sequence, person2}) => gql`mutation { Sequence { post_next_person(input:{id:"${sequence.id}", sourceId:"${person2.id}"}) { id } } }`,
        ({sequence, concept1}) => gql`mutation { Sequence { post_next_concept(input:{id:"${sequence.id}", sourceId:"${concept1.id}"}) { id } } }`,
        ({sequence, concept2}) => gql`mutation { Sequence { post_next_concept(input:{id:"${sequence.id}", sourceId:"${concept2.id}"}) { id } } }`,
      ],
      query: ({sequence}) => gql`query {
        Sequence(id:"${sequence.id}") {
          title
          class { title }
          head { type entity { title } next { type entity { title } } }
          tail { type entity { title } previous { type entity { title } } }
          elements { type entity { title } }
        }
      }`
    });
  });
});

// title
// class { title }
// spaces { title }
// head { type next { type } entity { id } }
// tail { type next { type } }

// classId
// tail { type sequence { title } entity { title } previous { entity { title } } }
// elements { type sequence { title } entity { title } next { entity { title } } previous { entity { title } } }