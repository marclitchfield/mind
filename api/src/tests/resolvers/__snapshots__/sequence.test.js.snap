// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sequence tests create sequence 1`] = `
Object {
  "Sequence": Object {
    "__typename": "Sequence",
    "class": Object {
      "__typename": "Concept",
      "title": "sequence type",
    },
    "elements": Array [
      Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 1",
        },
        "type": "Person",
      },
      Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 2",
        },
        "type": "Person",
      },
      Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 3",
        },
        "type": "Concept",
      },
      Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 4",
        },
        "type": "Concept",
      },
    ],
    "head": Object {
      "__typename": "SequenceElement",
      "entity": Object {
        "__typename": "Entity",
        "title": "element 1",
      },
      "next": Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 2",
        },
        "type": "Person",
      },
      "type": "Person",
    },
    "tail": Object {
      "__typename": "SequenceElement",
      "entity": Object {
        "__typename": "Entity",
        "title": "element 4",
      },
      "previous": Object {
        "__typename": "SequenceElement",
        "entity": Object {
          "__typename": "Entity",
          "title": "element 3",
        },
        "type": "Concept",
      },
      "type": "Concept",
    },
    "title": "test sequence",
  },
}
`;
